evolver/hardware/standard/led.py

```diff
@@ -34,3 +34,8 @@ def from_brightness(brightness):
         with self.serial as comm:
             comm.communicate(SerialData(addr=self.addr, data=cmd))
         self.committed = inputs
+
+    def abort(self):
+        cmd = [b"0"] * self.slots
+        with self.serial as comm:
+            comm.communicate(SerialData(addr=self.addr, data=cmd))
```

Comment:
I believe the above is what it takes to abort the operation of the device. Do you mean like naming it `off` since that would look less alarming if called upon initialization (if we do so in the future)?

Reply:
No, I meant that we both mentioned in #197 it might be desirable to exec these commands upon initial startup, in which case it's not really an "abort" but more of a ground/rest/init/off state.

Technically it's not aborting anything as we have to actually send a commit to command the hardware to some state. ``Evolver.abort`` is an "abort" but ``HardwareDriver.abort`` isn't.

Either way, I didn't mean anything complicated just something like the following:
```python
class HardwareDriver(...):
  @abstractmethod
  def init(self, *args, **kwargs):  # could be called "reset"
     """ Implement to command hardware to it's ground/off state. This could be called upon startup/abort."""
    ...
   
  def abort(self):
    self.init()
```

Reply:
I've renamed driver method to `off` and added docstring for method expectations. I will defer doing anything with startup in this PR.

Reply:
>Actually the device abort does send the command directly, no commit required.

I only meant that it still has to send a command period and not _just_ block subsequent commands. That these hardware methods don't alter the config, they themselves don't "block" future commanding at all etc, and that they are in fact completely oblivious to this concept (at least they _should_ be). They only alter the hardware to some specific state, a state in this instance that is encoded in the semantics of this function name - which isn't "abort".

>"abort" speaks to the immediacy of it though...

Agreed, and ``evolver.device.Evolver.abort`` is cool to be named such as _it_ does actually "abort" subsequent control etc, though there's still scope for argument around the fact that it still pings the hardware and reads data so doesn't completely "abort", see #199  #196.

>"abort" speaks to the immediacy of it though...

At the same time, however, this isn't entirely true due to the significant (1-2s) latency per call and the fact that these are turned off in sequence rather than proposals being ``set`` and committed all at once.